src/poetry/utils/password_manager.py

```diff
@@ -173,6 +174,25 @@ def set_pypi_token(self, name: str, token: str) -> None:
             self.keyring.set_password(name, "__token__", token)
 
     def get_pypi_token(self, name: str) -> str | None:
+        """Get PYPI token.
+
+        First checks the enviroment variables for a token,
+        then the configured username/password and last the
+        available keyring.
+
+
+        Parameters
+        ----------
+        name
+            PYPI username.
+
+        Returns
+        -------
+            Returns a token as a string if found, otherwise None.
+        """
+        if os.getenv("POETRY_PYPI_TOKEN_PYPI"):
+            return os.getenv("POETRY_PYPI_TOKEN_PYPI")
```

Comment:
... and therefore the environment variable that you want is `POETRY_PYPI_TOKEN_NAME` where `NAME` is `name.upper().replace("-", "_")`

Reply:
but maybe it would be simpler to rearrange this function as:

```
unconditionally try self._config.get()
(that will do the environment variable stuff)
if it succeeds, we're done

if keyring is available:
  try the key ring stuff
```

Reply:
> but maybe it would be simpler to rearrange this function as:
> 
> ```
> unconditionally try self._config.get()
> (that will do the environment variable stuff)
> if it succeeds, we're done
> 
> if keyring is available:
>   try the key ring stuff
> ```

I support this opinion.

Reply:
would something like this be any good then?:

```python
    def get_pypi_token(self, name: str) -> str | None:
        token = self._config.get(f"pypi-token.{name}")
        return (
            token if token is not None else self.keyring.get_password(name, "__token__")
        )
```

Because from what I understand `self._config.get` does check the environment variable?

Reply:
Looks good to me.

Reply:
I changed it, and fixed the unit tests. Please check both